cli/internal/kubernetes/upgrade.go

```diff
@@ -96,21 +97,37 @@ type Upgrader struct {
 	upgradeID  string
 }
 
+// NewUpgradeID returns a new upgrade ID.
+// is private function to avoid construction outside this function.
+//
+//nolint:all
```

Comment:
the comment is above ;) I wanted to enforce no usage outside this function

Reply:
I don't think hitting the constructor with a `nolint:all` because the returned type is private is the correct way of going about this.
We have plenty data types that use constructors (and should only created using those Constructors).

Enforcing the creation of an `upgradeID` using `NewUpgradeID` is already given by the fields of the data type being private.

Reply:
So you suggest to export the type? @daniel-weisse 
Don't mean to be pedantic
```
	upgradeID := kubernetes.UpgradeID{"g"}
```
this would still work but I guess it's obvious that this is wrong usage?